compiler/src/lnntoamm/Expr.ts

```diff
@@ -543,7 +565,9 @@ class Call extends Expr {
     }
     // first reduction
     const argTys = args.map((arg) => arg.ty);
-    fns = Fn.select(fns, argTys, metadata.scope);
+    const selFns = FunctionType.matrixSelect(fns, argTys, metadata.scope);
+    fns = selFns.map(([fn, _ty]) => fn);
```

Comment:
Nit: couldn't this just be `fns = selFns.map(selFn => selFn[0]);`

Doing function argument-level destructuring and assignment to an unused variable seems like a performance penalty that will affect interpreter mode heavily and may still trip up the JIT on being able to fully optimize back down to the form I wrote above?